codec-http/src/main/java/io/netty5/handler/codec/http/headers/DefaultHttpHeaders.java

```diff
@@ -0,0 +1,772 @@
+/*
+ * Copyright 2022 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+/*
+ * Copyright © 2018, 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.netty5.handler.codec.http.headers;
+
+import io.netty5.util.AsciiString;
+import io.netty5.util.internal.UnstableApi;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import static io.netty5.handler.codec.http.HttpHeaderNames.COOKIE;
+import static io.netty5.handler.codec.http.HttpHeaderNames.SET_COOKIE;
+import static io.netty5.handler.codec.http.headers.DefaultHttpSetCookie.parseSetCookie;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.DEFAULT_HEADER_FILTER;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.domainMatches;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.isSetCookieNameMatches;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.parseCookiePair;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.pathMatches;
+import static io.netty5.handler.codec.http.headers.HeaderUtils.validateToken;
+import static io.netty5.util.AsciiString.contentEquals;
+import static io.netty5.util.AsciiString.contentEqualsIgnoreCase;
+import static java.util.Collections.emptyIterator;
+
+/**
+ * Default implementation of {@link HttpHeaders}.
+ */
+@UnstableApi
```

Comment:
I think we can remove all the `@UnstableApi` as as soon as we release it must be stable

Reply:
My idea was to only expose the interface, which has static factory methods, but the `DefaultHttp2Headers` implementation need to extend it, so maybe it should just be public API.